src/main/java/co/cask/gcp/speech/SpeechToTextTransform.java

```diff
@@ -0,0 +1,383 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.gcp.speech;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.StageSubmitterContext;
+import co.cask.cdap.etl.api.Transform;
+import co.cask.cdap.etl.api.TransformContext;
+import co.cask.gcp.common.GCPConfig;
+import co.cask.gcp.common.GCPUtils;
+import com.google.cloud.speech.v1.RecognitionAudio;
+import com.google.cloud.speech.v1.RecognitionConfig;
+import com.google.cloud.speech.v1.RecognizeResponse;
+import com.google.cloud.speech.v1.SpeechClient;
+import com.google.cloud.speech.v1.SpeechRecognitionAlternative;
+import com.google.cloud.speech.v1.SpeechRecognitionResult;
+import com.google.cloud.speech.v1.SpeechSettings;
+import com.google.protobuf.ByteString;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import javax.ws.rs.Path;
+
+/**
+ * Google Cloud Speech-To-Text Transform which transcripts audio
+ */
+@Plugin(type = Transform.PLUGIN_TYPE)
+@Name(SpeechToTextTransform.NAME)
+@Description(SpeechToTextTransform.DESCRIPTION)
+public class SpeechToTextTransform extends Transform<StructuredRecord, StructuredRecord> {
+  public static final String NAME = "SpeechToText";
+  public static final String DESCRIPTION = "Converts audio files to text by applying powerful neural network models.";
+  private SpeechTransformConfig config;
+  private Schema outputSchema = null;
+  private Schema transcriptionPartsSchema = null;
+  private RecognitionConfig recognitionConfig;
+  private SpeechClient speech;
+
+  @Override
+  public void configurePipeline(PipelineConfigurer configurer) throws IllegalArgumentException {
+    super.configurePipeline(configurer);
+    config.validate(configurer.getStageConfigurer().getInputSchema());
+  }
+
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+    config.validate(context.getInputSchema());
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+    if (context.getInputSchema() != null) {
+      outputSchema = getSchema(context.getInputSchema(), config.getPartsField(), config.getTextField());
+    }
+    if (config.getPartsField() != null) {
+      transcriptionPartsSchema = Schema.recordOf("record", transcriptWithConfidenceSchema(config.getPartsField()));
```

Comment:
this shouldn't be a record, it should just be what you get from transcriptWithConfidenceSchema()